gridcapa-core-cc-app/src/test/java/com/farao_community/farao/gridcapa_core_cc/app/preprocessing/RaoParametersServiceTest.java

```diff
@@ -49,4 +60,278 @@ void checkRaoParametersFilledCorrectly() {
+    }
+
+    @Test
+    void loopFlowCountriesWithBadCountryTest() {
+        RequestMessage requestMessage = new RequestMessage();
+        Header header = new Header();
+        requestMessage.setHeader(header);
+        Property property = new Property();
+        property.setName("LF_CONSTRAINT_test");
+        property.setValue("true");
+        header.getProperty().add(property);
+        RaoParameters raoParameters = new RaoParameters();
+
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> RaoParametersService.setLoopFlowCountries(requestMessage, raoParameters)
+        );
+    }
+
+    @Test
+    void loopFlowCountriesFalseTest() {
+        RequestMessage requestMessage = new RequestMessage();
+        Header header = new Header();
+        requestMessage.setHeader(header);
+        Property property = new Property();
+        header.getProperty().add(property);
+        property.setName("LF_CONSTRAINT_FR");
+        property.setValue("false");
+        RaoParameters raoParameters = new RaoParameters();
+
+        RaoParametersService.setLoopFlowCountries(requestMessage, raoParameters);
+
+        LoopFlowParametersExtension loopFlowParametersExtension = raoParameters.getExtension(LoopFlowParametersExtension.class);
+        assertNotNull(loopFlowParametersExtension.getCountries());
+        assertEquals(0, loopFlowParametersExtension.getCountries().size());
+    }
+
+    @Test
+    void ptdfBoundariesWithExtensionTest() {
+        VirtualHubsConfiguration virtualHubsConfiguration = new VirtualHubsConfiguration();
+        RaoParameters raoParameters = new RaoParameters();
+        RelativeMarginsParametersExtension extension = new RelativeMarginsParametersExtension();
+        raoParameters.addExtension(RelativeMarginsParametersExtension.class, extension);
+        extension.setPtdfSumLowerBound(76.0);
+
+        RaoParametersService.setPtdfBoundaries(virtualHubsConfiguration, raoParameters);
+
+        RelativeMarginsParametersExtension rmpExtension = raoParameters.getExtension(RelativeMarginsParametersExtension.class);
+        assertEquals(76.0, rmpExtension.getPtdfSumLowerBound());
+    }
+
+    @Test
+    void ptdfBoundariesWithoutExtensionTest() {
+        VirtualHubsConfiguration virtualHubsConfiguration = new VirtualHubsConfiguration();
+        RaoParameters raoParameters = new RaoParameters();
+
+        RaoParametersService.setPtdfBoundaries(virtualHubsConfiguration, raoParameters);
+
+        RelativeMarginsParametersExtension extension = raoParameters.getExtension(RelativeMarginsParametersExtension.class);
+        assertTrue(extension.getPtdfSumLowerBound() < 1.0);
+    }
+
+    @Test
+    void ptdfBoundariesFromBorderDirectionsTest() {
+        VirtualHubsConfiguration virtualHubsConfiguration = new VirtualHubsConfiguration();
+        virtualHubsConfiguration.addMarketArea(new MarketArea("FR", "10YFR-RTE------C", true, false));
```

Comment:
Given the duplication of the country and the eic in several tests, i propose that we declare this variable as constant.